lib/auto_attach.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+"""
+Perform auto-attach operation
+
+On Ubuntu Pro machines, we try to perform an auto-attach operation
+on first boot. This happens through a systemd unit that triggers
+that executes this script on every boot. However, if we detect
+that cloud-init has user data related to ua, we cannot run
+auto-attach here, since cloud-init will drive this operation on
+their side.
+"""
+import logging
+import sys
+
+from uaclient.cli import action_auto_attach, setup_logging
+from uaclient.config import UAConfig
+from uaclient.exceptions import AlreadyAttachedOnPROError
+
+try:
+    import cloudinit.stages as ci_stages  # type: ignore
+except ImportError:
```

Comment:
Is the intent here to support images which may not contain cloud-init? maybe some container builds or maybe desktop images? `cloudinit.stages` should always exist on most server images everywhere, and any desktop live install images. But there is the case for desktop not-live-installer images that maybe this import handling is for. If that's the case, can we add a comment here about what image types this import handling is being used for.

Reply:
I guess additionally do we have a use case for desktop images or containers running auto-attach where we'd be concerned about even executing this code on those image types?

Reply:
Yes, the general idea here is to avoid potential issue for any environment that lacks cloud-init and have the pro package installed, even by accident. However, I don't know a real use case where that happens

Reply:
I don't think we currently have a use-case for non-cloud images. However, cloud-init is not (and cannot be) an explicit dependency, and so I think we should be defensive when relying on it.